python/pyarrow/scalar.pxi

```diff
@@ -731,6 +731,57 @@ cdef class LargeListValue(ArrayValue):
         return result
 
 
+cdef class FixedSizeListValue(ArrayValue):
+    """
+    Concrete class for fixed size list array elements.
+    """
+
+    def __len__(self):
+        """
+        Return the number of values.
+        """
+        return self.length()
+
+    def __getitem__(self, i):
+        """
+        Return the value at the given index.
+        """
+        return self.getitem(_normalize_index(i, self.length()))
+
+    def __iter__(self):
+        """
+        Iterate over this element's values.
+        """
+        for i in range(len(self)):
+            yield self.getitem(i)
+        raise StopIteration
+
+    cdef void _set_array(self, const shared_ptr[CArray]& sp_array):
+        self.sp_array = sp_array
+        self.ap = <CFixedSizeListArray*> sp_array.get()
+        self.value_type = pyarrow_wrap_data_type(self.ap.value_type())
+
+    cdef getitem(self, int64_t i):
+        cdef int64_t j = self.ap.value_offset(self.index) + i
+        return box_scalar(self.value_type, self.ap.values(), j)
```

Comment:
not familiar with what box_scalar does but it seems strange to be boxing a non-scalar type list?

Reply:
> not familiar with what box_scalar does but it seems strange to be boxing a non-scalar type list?

This is getting one element of the list value (not boxing the full list value), which typically is a scalar, unless you have nested lists (so `self` is a single list of the list array, and this method is then getting one element of that list)